src/main/java/seedu/address/model/jobapplication/JobApplication.java

```diff
@@ -0,0 +1,152 @@
+package seedu.address.model.jobapplication;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.time.LocalDateTime;
+import java.util.Objects;
+
+import seedu.address.commons.util.ToStringBuilder;
+
+/**
+ * Represents a Job Application.
+ * Guarantees: details are present and not null, field values are validated, immutable.
+ */
+public class JobApplication {
+
+    /**
+     * Represents the status of a job application.
+     */
+    public enum Status {
+        APPLIED,
+        INPROGRESS,
+        REJECTED
+    }
+
+    // Identity fields
+    private final String companyName;
+    private final String role;
+
+    // Data fields
+    private final LocalDateTime deadline;
+    private final Status status;
+
+    /**
+     * Constructs a JobApplication.
+     * Every field must be present and not null.
+     *
+     * @param companyName The name of the company.
+     * @param role The job role applied for.
+     * @param deadline The application deadline.
+     * @param status The current status of the application.
+     */
+    public JobApplication(String companyName, String role, LocalDateTime deadline, Status status) {
+        requireAllNonNull(companyName, role, deadline, status);
+        this.companyName = companyName;
+        this.role = role;
+        this.deadline = deadline;
+        this.status = status;
+    }
+
+    /**
+     * Returns the company name.
+     *
+     * @return The name of the company.
+     */
+    public String getCompanyName() {
+        return companyName;
+    }
+
+    /**
+     * Returns the job role.
+     *
+     * @return The job role applied for.
+     */
+    public String getRole() {
+        return role;
+    }
+
+    /**
+     * Returns the application deadline.
+     *
+     * @return The deadline for the job application.
+     */
+    public LocalDateTime getDeadline() {
+        return deadline;
+    }
+
+    /**
+     * Returns the application status.
+     *
+     * @return The current status of the application.
+     */
+    public Status getStatus() {
+        return status;
+    }
+
+    /**
+     * Returns true if both job applications have the same company name and role.
+     * This defines a weaker notion of equality between two job applications.
+     *
+     * @param otherJobApplication The other job application to compare with.
+     * @return True if both job applications have the same company name and role.
+     */
+    public boolean isSameJobApplication(JobApplication otherJobApplication) {
+        if (otherJobApplication == this) {
+            return true;
+        }
+
+        return otherJobApplication != null
+                && otherJobApplication.getCompanyName().equals(getCompanyName())
```

Comment:
can we standarise the use of getters or direct access to member attributes e.g. `getCompanyName()` found in `isSameJobApplication` and `companyName` in `equals`